File: App.client/projects/proctor-app/scripts/models/notification.ts
import { User, RelatedEntityType } from './user';

export class Notification {
  id: number;
  userId: number;
  title: string;
  message: string;
  isRead: boolean;
  createdAt: string;
  relatedEntityType: string;
  relatedEntityId?: number | null;
  actionUrl?: string | null;
  createdOn: string;
  modifiedOn: string;
  user?: User | null;

  constructor() {
    this.id = 0;
    this.userId = 0;
    this.title = this.message = '';
    this.isRead = false;
    this.createdAt = new Date().toISOString();
    this.relatedEntityType = RelatedEntityType.Complaint;
    this.relatedEntityId = this.actionUrl = null;
    this.createdOn = this.modifiedOn = '';
  }
}

File: App.Models/Notification.cs
using App.Core.Models;

namespace App.Models
{
    public class Notification : IAuditableEntity
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Message { get; set; } = string.Empty;
        public bool IsRead { get; set; } = false;
        public DateTime CreatedAt { get; set; }
        public RelatedEntityType RelatedEntityType { get; set; }
        public int? RelatedEntityId { get; set; }
        public string? ActionUrl { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime ModifiedOn { get; set; }

        // Navigation properties
        public virtual User User { get; set; } = null!;
    }
}


